### PR TITLE
Allow bitarray to handle shifts by a negative value

### PR DESCRIPTION
**Pull request by dbeach24**
Fix for #16706.
This allows left/right shifting a bitarray by a negative value.

### CODE REVIEW

**Comment by dbeach24** [comment]
@eschnett Could you please verify that this does what you wanted?

**Comment by dbeach24** [comment]
Is this merge-worthy?

**Comment by tkelman** [comment]
LGTM

File: test/bitarray.jl [hunk 1]
```diff
@@ -876,7 +876,9 @@ end
 b1 = bitrand(v1)
 for m = [rand(1:v1)-1 0 1 63 64 65 191 192 193 v1-1]
     @test isequal(b1 << m, [ b1[m+1:end]; falses(m) ])
+    @test isequal(b1 << -m, b1 >> m)
     @test isequal(b1 >>> m, [ falses(m); b1[1:end-m] ])
+    @test isequal(b1 >>> -m, b1 << m)
```
Comment (rfourquet) [inline test/bitarray.jl]: Maybe, for completeness, add a test for `b1 >> -m` ?

**Comment by rfourquet** [comment]
LGTM too.

**Comment by rfourquet** [comment]
Also while you are at it, you could update the docs for shift operations (e.g. the doc for `>>` implicitly assumes that the LHS is an integer type (with a sign)). Otherwise let's open a doc issue for that.

File: test/bitarray.jl (hunk 1 above)
Comment (dbeach24) [inline test/bitarray.jl]: Will do

**Comment by dbeach24** [comment]
@rfourquet Not clear on how to update the docs.  The docstrings for the shift operators are found in base/operators.jl, where the methods are defined for integer types.  I don't want to change the signature of these functions (since they are specific to Integer).

Can you clarify what I'm supposed to do here?

**Comment by rfourquet** [comment]
I think it would be enough to attach additional docstrings to the shift methods acting on bitarrays.

**Comment by kshyatt** [comment]
I'm bumping this - can we merge this as-is and deal with the docs later? Alternatively, can someone guide @dbeach24 through the process of adding the docstrings?

**Comment by rfourquet** [comment]
I think it can, but would be nice to at least open a doc issue.

**Comment by tkelman** [comment]
replaced by #19760

**Pull request closed without merging**